### PR TITLE
feat: add to cart, cart page

### PR DESCRIPTION
**Pull request by yjkimjunior**
implements #188

follow up PR required for adding store.js to automatically subscribe to localstorage.

fsz4RVHsmh

cart only shows 2 items because it won't add duplicates. There shouldn't be duplicates in the table in the first place though, see: https://github.com/paritytech/Nomidot/issues/206

Blocknumber is fetching because nodewatcher went haywire. restarted this morning, and it's flying through. should be synced by evening if it doesn't go tup again.

### CODE REVIEW

File: front/gatsby/src/apollo.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const wsLink = new WebSocketLink({
   options: {
     reconnect: true,
   },
-  webSocketImpl: process.env.NODE_ENV === 'production' ? ws : WebSocket, // ws does not work in the browser. Browser clients must use the native WebSocket object. But during production SSR, we must use ws.
+  webSocketImpl: typeof window === 'undefined' ? ws : window.WebSocket, // ws does not work in the browser. Browser clients must use the native WebSocket object. But during production SSR, we must use ws.
```
Comment (yjkimjunior) [inline front/gatsby/src/apollo.ts]: annoyingly this has very inconsistent behavior in dev mode...

**Comment by yjkimjunior** [comment]
ok i'm just going to merge this and keep moving forrward if no one has anything to say....

File: front/gatsby/src/apollo.ts (hunk 1 above)
Comment (amaurymartiny) [reply front/gatsby/src/apollo.ts]: could use IS_SSR

File: front/gatsby/src/components/Layout/Header/Header.tsx [hunk 2]
```diff
@@ -192,54 +40,71 @@ export function Header(): React.ReactElement {
   }, [fetchAccounts]);
 
   useEffect(() => {
+    const count = getCartItemsCount();
+    setNumberOfItemsInCart(count);
     handleLogin();
+
+    // FIXME: use store.js for window.addEventListener('storage', updateCartItems);
```
Comment (amaurymartiny) [inline front/gatsby/src/components/Layout/Header/Header.tsx]: "there's a hook for everything" https://usehooks.com/useLocalStorage/ (just fyi, not saying it's better than window.addEventListener)

Comment (amaurymartiny) [reply front/gatsby/src/components/Layout/Header/Header.tsx]: i think you would need to put all cart items under one key, aka an array inside one locastroage key inside of `n` keys. But should be doable

**Pull request merged**